### PR TITLE
Se implementan nuevas funciones, se elimina código innecesario, se ut…

### PR DESCRIPTION
**Pull request by bryanm92s**
**New features are implemented:**

1. Unnecessary code is removed
2. Binding is used
3. MVVM architecture is implemented in SignUp and Login
4. Coroutines are implemented
5. Design changes are made
6. Variable names are renamed from Spanish to English
7. Dimens and Strings are used

coroutines
englishFragments
home
mvvm
play
profile
search

https://user-images.githubusercontent.com/21268408/176741133-69cdd8f6-f420-4b42-b2d5-69f1af6b1bb9.mp4

### CODE REVIEW

**Comment by bryanm92s** [comment]
@jrojas25

File: app/src/main/java/com/example/apppractica/register/RegisterFragment.kt [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+        registerViewModel.navigateto.observe(viewLifecycleOwner, Observer { hasFinished ->
+            if (hasFinished == true) {
+                Log.i("MYTAG", "insidi observe")
+                displayLogin()
+                registerViewModel.doneNavigating()
+            }
+        })
+
+        registerViewModel.userDetailsLiveData.observe(viewLifecycleOwner, Observer {
+            Log.i("MYTAG", it.toString() + "000000000000000000000000")
```
Comment (jrojas25) [inline app/src/main/java/com/example/apppractica/register/RegisterFragment.kt:54]: what is this for?

File: app/src/main/java/com/example/apppractica/register/RegisterFragment.kt [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+                registerViewModel.donetoast()
+            }
+        })
+        
+        // Error Username
+        registerViewModel.errortoastUsername.observe(viewLifecycleOwner, Observer { hasError ->
+            if (hasError == true) {
+                Toast.makeText(requireContext(), "UserName Already taken", Toast.LENGTH_SHORT)
+                    .show()
+                registerViewModel.donetoastUserName()
```
Comment (jrojas25) [inline app/src/main/java/com/example/apppractica/register/RegisterFragment.kt:71]: why this should go back to the viewmodel once it gets an update from it?

File: app/src/main/java/com/example/apppractica/register/RegisterFragment.kt [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+            }
+        })
+
+        // Error Email
+        registerViewModel.errortoastEmail.observe(viewLifecycleOwner, Observer { hasError ->
+            if (hasError == true) {
+                Toast.makeText(requireContext(), "Enter a valid email", Toast.LENGTH_SHORT).show()
+                registerViewModel.donetoastEmail()
```
Comment (jrojas25) [inline app/src/main/java/com/example/apppractica/register/RegisterFragment.kt:79]: why this should go back to the viewmodel once it gets an update from it?

File: app/src/main/java/com/example/apppractica/register/RegisterFragment.kt [hunk 4]
```diff
@@ -0,0 +1,105 @@
...
+
+        // Error Password
+        registerViewModel.errortoastPassword.observe(viewLifecycleOwner, Observer { hasError ->
+            if (hasError == true) {
+                Toast.makeText(
+                    requireContext(),
+                    "Password must contain at least 8 characters",
+                    Toast.LENGTH_SHORT
+                ).show()
+                registerViewModel.donetoastPassword()
```
Comment (jrojas25) [inline app/src/main/java/com/example/apppractica/register/RegisterFragment.kt:91]: why this should go back to the viewmodel once it gets an update from it?

File: app/src/main/java/com/example/apppractica/register/RegisterViewModel.kt [hunk 5]
```diff
@@ -0,0 +1,141 @@
...
+    fun sumbitButton() {
+        Log.i("MYTAG", "Inside SUBMIT BUTTON")
+        if (inputUsername.value == null || inputEmail.value == null || inputPassword.value == null) {
+            _errorToast.value = true
+        } else if (!PatternsCompat.EMAIL_ADDRESS.matcher(inputEmail.value.toString()).matches()) {
+            _errorToastEmail.value = true
+        } else if (inputPassword.value!!.length < 8) {
+            _errorToastPassword.value = true
+        } else {
+            uiScope.launch {
```
Comment (jrojas25) [inline app/src/main/java/com/example/apppractica/register/RegisterViewModel.kt:77]: when doing tasks in background avoid using the UI thread

File: app/src/main/java/com/example/apppractica/register/RegisterViewModel.kt [hunk 6]
```diff
@@ -0,0 +1,141 @@
...
+
+    private fun insert(user: RegisterEntity): Job = viewModelScope.launch {
+        repository.insert(user)
+    }
+
+    override fun removeOnPropertyChangedCallback(callback: Observable.OnPropertyChangedCallback?) {
+    }
+
+    override fun addOnPropertyChangedCallback(callback: Observable.OnPropertyChangedCallback?) {
+    }
```
Comment (jrojas25) [inline app/src/main/java/com/example/apppractica/register/RegisterViewModel.kt:134]: what is this for?

File: app/src/main/res/layout/activity_main.xml [hunk 7]
```diff
@@ -4,36 +4,35 @@
...
+            app:layout_constraintTop_toTopOf="parent"
+            app:navGraph="@navigation/navigation" />
 
+        <com.google.android.material.bottomnavigation.BottomNavigationView
+            android:id="@+id/bottomNavigationView"
+            android:layout_width="match_parent"
+            android:layout_height="wrap_content"
+            android:background="@drawable/round_corners"
+            android:paddingTop="10dp"
+            android:paddingBottom="10dp"
```
Comment (jrojas25) [inline app/src/main/res/layout/activity_main.xml:28]: use dimens (and check the same for ALL of the layouts)

**Review by jrojas25 — CHANGES_REQUESTED** [review]

**Pull request closed without merging**